build/ci_utils/src/goodies/shader_tools.rs

```diff
@@ -0,0 +1,73 @@
+use crate::prelude::*;
+
+use crate::cache::goodie;
+use crate::cache::Cache;
+use crate::cache::Goodie;
+use crate::env::known::PATH;
+use crate::programs::shaderc::Glslc;
+use crate::programs::shaderc::SpirvOpt;
+use crate::programs::spirv_cross::SpirvCross;
+
+#[derive(Clone, Copy, Debug, Default)]
+pub struct ShaderTools;
+
+impl Goodie for ShaderTools {
```

Comment:
In general something that can be obtained through IO action, cached and enabled through the changes to the global state (mostly environment). It will be heavily refactored when offline mode is implemented.